### PR TITLE
DEV-1745 Pre-commit Updates

### PR DESCRIPTION
**Pull request by njharman**
#### What's this PR do?
Update pre-commit tools and fix existing issues reported by pre-commit.
Update workflow action versions and get CI all passing.

Ran pre-commit autoupdate
Ran pre-commit --all-files

The files changed:
- .pre-commit-config.yaml
- .github/workflows/pre-comimt.yml
- .github/workflows/test-js.yml
- .github/workflows/test-python.yml
- manual fixes to Dockerfile (I guessed at proper FROM base... usage)
- Rest of changes were made by line ending and trailing whitespace checkers

#### Why are we doing this? How does it help us?
More current checks. Part of getting py3.9/3.10 in CI.

#### How should this be manually tested?
`pre-commit --all-files`  should not report issues
CI should pass cleanly

#### Have automated unit tests been added?
N/A

#### How should this change be communicated to end users?
N/A

#### Are there any smells or added technical debt to note?
No? 

#### Has this been documented? If so, where?
N/A.

#### What are the relevant tickets?
[DEV-1745]

#### Do any changes need to be made before deployment to production (adding environment variables, for example)?
No.

#### Are there next steps? If so, what? Have tickets been opened for them?
Devs should configure their editors to auto delete trailing whitespace (or at least highlight it for manual deletion)

DEV-1751 to enable Talisman

pylint is not enabled in pre-commit.yaml

There is duplication between pre-commit.yml and test-python.yml.  test-python uses older method of using poetry. Did not want to spend anymore time on ticket to resolve these.

[DEV-1745]: https://news-revenue-hub.atlassian.net/browse/DEV-1745?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -57,7 +54,7 @@ RUN mkdir /code/
 WORKDIR /code/
 ADD . /code/
 
-FROM base AS deploy
+FROM base:3.9-slim  AS deploy
```
Comment (x110dc) [inline Dockerfile:57]: Why was this changed? It makes the Docker image fail to build.

**Review by x110dc — CHANGES_REQUESTED** [review]
One change requested.

File: Dockerfile (hunk 1 above)
Comment (njharman) [reply Dockerfile:57]: The docker linter complained about having "non-pegged" FROM xxx.

I don't know Docker very well. I guessed at 3.9-slim cause that is what  FROM .... as base line had. linter didn't want :latest.  I didn't know how else to say "the one we just created, right here, figure it out computer!"

Comment (njharman) [reply Dockerfile]: commented out and punted on See DEV-1763

**Review by x110dc — APPROVED** [review]
LGTM.

**Pull request merged**